src/html_json.js

```diff
@@ -213,29 +231,25 @@ module.exports.main = async (context, action) => {
   };
 
   const config = await loadConfig();
-  const docs = [];
 
-  const htmlIndices = await fetchHTML({
-    owner, repo, ref, path, logger: action.logger,
-  }, config.indices);
-
-  Object.entries(htmlIndices)
-    .filter(([, { response }]) => response !== null)
-    .forEach(([name, { index, response }]) => {
-      const { document } = new JSDOM(response).window;
-      if (index.group) {
-        docs.push(...indexGroup(path, document, index));
-      } else {
-        docs.push({
-          [name]: indexSingle(path, document, index, logger),
-        });
-      }
-    });
-  return {
-    response: {
-      body: {
-        docs,
-      },
-    },
-  };
+  try {
+    const htmlIndices = await fetchHTML({
+      owner, repo, ref, path, log: action.logger,
+    }, config.indices);
+
+    const body = {};
+    Object.entries(htmlIndices)
+      .reduce((prev, [name, { index, result: { error, response } }]) => {
+        if (error) {
+          body[name] = { error };
+        } else {
+          body[name] = { docs: evaluateHtml(response, path, index, log) };
+        }
+        return body;
+      }, body);
+    return { response: { body } };
```

Comment:
This goes with https://github.com/adobe/helix-index-files/pull/162: the body of the response contains one entry per index definition, which again contains a `docs` array.